ui_framework/components/tabs/tabs.js

```diff
@@ -0,0 +1,116 @@
+import React, { cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { KuiTab } from './tab';
+
+export class KuiTabs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedTabIx: this.getStartSelectedIx()
+    };
+  }
+
+  getStartSelectedIx() {
+    const tabsLength = this.props.tabs.length;
+    const selectedTabIx = this.props.selectedTabIx;
+
+    if (tabsLength > 0) {
+      return selectedTabIx ? selectedTabIx : 0;
+    } else {
+      return undefined;
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selectedTabIx && (this.state.selectedTabIx !== nextProps.selectedTabIx)) {
+      this.setState({ selectedTabIx:nextProps.selectedTabIx });
+    }
+  }
+
+  render() {
+    const {
+      tabs,
+      onSelectedTabChanged, //eslint-disable-line
+      className,
+      ...rest
+    } = this.props;
+    const classes = classNames('kuiTabs', className);
+    const children = this.getChildren(tabs);
+
+    return (
+      <div
+        className={classes}
+        {...rest}
+      >
+        {children}
+      </div>
+    );
+  }
+
+  getChildren(tabs) {
+    return tabs.map((tab,ix)=>{
+      const props = {
+        isSelected: ix === this.state.selectedTabIx,
+        onClick: ()=>this.tabClicked(ix),
+        key: ix
+      };
+
+      return cloneElement(tab,props);
```

Comment:
I really like your approach with the `TabsCompact` component. I think we should use it! Let's make it a little less restrictive so that it allows you to provide a `node` type, instead of string. This will allow us to make adjustments to the way tabs appear by adding new components (e.g. a `<KuiTabIcon>` component), which we can provide as children, instead of needing to change the implementation of the Tabs component itself.

So this method will become:

```jsx
return tabs.map((tab, index) => (
  <KuiTab
    key={index}
    onClick={() => this.props.onSelectedTabChanged(index)}
    isSelected={index === this.props.selectedTabIndex}
  >
    {tab}
  </KuiTab>
));
```